Sources/News.php

```diff
@@ -1849,4 +2129,758 @@ function getXmlProfile($xml_format)
+						'content' => !empty($row['modified_time']) ? $smcFunc['htmlspecialchars'](strip_tags(timeformat($row['modified_time']))) : null,
+					),
+					array(
+						'tag' => 'modified_by',
+						'attributes' => !empty($row['modified_name']) ? array('title' => $txt['modified_by']) : null,
+						'content' => !empty($row['modified_name']) ? $row['modified_name'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'modified_reason',
+						'attributes' => !empty($row['modified_reason']) ? array('title' => $txt['reason_for_edit']) : null,
+						'content' => !empty($row['modified_reason']) ? $row['modified_reason'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'likes',
+						'attributes' => array('title' => $txt['likes']),
+						'content' => $row['likes'],
+					),
+					array(
+						'tag' => 'attachments',
+						'attributes' => array('title' => $txt['attachments']),
+						'content' => $attachments,
+					),
+				),
+			);
+		}
+	}
+	$smcFunc['db_free_result']($request);
+
+	// If we're in batch mode, make a note of our progress.
+	if (!empty($context['batch_mode']))
+	{
+		$context['batch_prev'] = (empty($context['batch_prev']) ? 0 : $context['batch_prev']) + count($data);
+
+		file_put_contents($cachedir . '/xml-batch-posts-' . $uid, implode(';', array($last, $context['batch_prev'], $context['batch_total'])));
+	}
+
+	return $data;
+}
+
+/**
+ * Get a user's personal messages.
+ * Only the user can do this, and no one else -- not even the admin!
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPMs($xml_format)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	// Personal messages are supposed to be private
+	if (isset($_GET['u']) && (int) $_GET['u'] != $user_info['id'])
+		return array();
+
+	// For batch mode, we need to know how many there are
+	if (!empty($context['batch_mode']) && empty($context['batch_total']))
+	{
+		$request = $smcFunc['db_query']('', '
+			SELECT COUNT(pm.id_pm)
+			FROM {db_prefix}personal_messages AS pm
+				INNER JOIN {db_prefix}pm_recipients AS pmr ON (pm.id_pm = pmr.id_pm)
+			WHERE (pm.id_member_from = {int:uid} OR pmr.id_member = {int:uid})',
+			array(
+				'uid' => $user_info['id'],
+			)
+		);
+		list($context['batch_total']) = $smcFunc['db_fetch_row']($request);
+		$smcFunc['db_free_result']($request);
+	}
+
+	$request = $smcFunc['db_query']('', '
+		SELECT pm.id_pm, pm.msgtime, pm.subject, pm.body, pm.id_member_from, pm.from_name, GROUP_CONCAT(pmr.id_member) AS id_members_to, GROUP_CONCAT(COALESCE(mem.real_name, mem.member_name)) AS to_names
```

Comment:
group_concat doesn't exists for integer fields on pg setups

Reply:
What do you suggest @albertlast  ?

Reply:
The postgres syntax is ``string_agg(some_column, ',')``
see https://www.postgresql.org/docs/11/sql-expressions.html

So you got two options change the agg function dynamical based on the database or
extend the install/upgrade script that group_concat get a alias on string_agg,
similiar stuff we already done here:
https://github.com/SimpleMachines/SMF2.1/blob/71a42263f3f8d413579934743d3e8df6555ea638/other/install_2-1_postgresql.sql#L9-L29